test/distributed/org/apache/cassandra/distributed/upgrade/ReplicationUpgradeTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.upgrade;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.dht.Murmur3Partitioner;
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
+import org.apache.cassandra.distributed.shared.Versions;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertRows;
+import static org.apache.cassandra.distributed.shared.AssertUtils.row;
+
+public class ReplicationUpgradeTest extends UpgradeTestBase
+{
+    @Test
+    public void testSimpleStrategy() throws Throwable
```

Comment:
This is looking really nice. Are we going to add some additional consistency levels? The problem with lower CLs is that we can't only make assumptions about the minimum number of written replicas.

Reply:
@adelapena Yeah, writing at the lower CLs doesn't seem to be super valuable if the point of all this is to make sure replication works properly. We just use ALL to make the tests deterministic, although I think I'll probably work out some additional tests elsewhere to cover missing hint delivery scenarios, if there are any. (I think we had identified a few in the [audit doc](https://docs.google.com/document/d/1yPbquhAALIkkTRMmyOv5cceD5N5sPFMB1O4iOd3O7FM/edit?disco=AAAAKxI9jj4).)

Reply:
Agree, we don't really need to test a lot of consistency levels here.